diff --git a/wpiutil/src/main/native/unix/StackTrace.cpp b/wpiutil/src/main/native/unix/StackTrace.cpp
index 3b6606533..c2e322b46 100644
--- a/wpiutil/src/main/native/unix/StackTrace.cpp
+++ b/wpiutil/src/main/native/unix/StackTrace.cpp
@@ -4,7 +4,9 @@
 
 #include "wpi/StackTrace.h"
 
+#ifndef __EMSCRIPTEN__
 #include <execinfo.h>
+#endif
 
 #include "wpi/Demangle.h"
 #include "wpi/SmallString.h"
@@ -14,6 +16,9 @@
 namespace wpi {
 
 std::string GetStackTraceDefault(int offset) {
+#ifdef __EMSCRIPTEN__
+  return "";
+#else
   void* stackTrace[128];
   int stackSize = backtrace(stackTrace, 128);
   char** mangledSymbols = backtrace_symbols(stackTrace, stackSize);
@@ -36,6 +41,7 @@ std::string GetStackTraceDefault(int offset) {
   std::free(mangledSymbols);
 
   return std::string{trace.str()};
+#endif
 }
 
 }  // namespace wpi
