diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79df35974..2573a5059 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -213,8 +213,6 @@ endif()
 
 find_package(LIBSSH 0.7.1)
 
-find_package(Protobuf REQUIRED)
-
 set(APRILTAG_DEP_REPLACE "find_dependency(apriltag)")
 set(CAMERASERVER_DEP_REPLACE_IMPL "find_dependency(cameraserver)")
 set(CSCORE_DEP_REPLACE_IMPL "find_dependency(cscore)")
diff --git a/wpimath/CMakeLists.txt b/wpimath/CMakeLists.txt
index 3c9c7dcc9..1032e2f78 100644
--- a/wpimath/CMakeLists.txt
+++ b/wpimath/CMakeLists.txt
@@ -5,18 +5,6 @@ include(CompileWarnings)
 include(AddTest)
 include(DownloadAndCheck)
 
-# workaround for makefiles - for some reason parent directories aren't created.
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/protobuf")
-file(GLOB wpimath_proto_src src/main/proto/*.proto)
-protobuf_generate_cpp(
-    WPIMATH_PROTO_SRCS
-    WPIMATH_PROTO_HDRS
-    PROTOC_OUT_DIR
-    "${CMAKE_CURRENT_BINARY_DIR}/protobuf"
-    PROTOS
-    ${wpimath_proto_src}
-)
-
 file(
     GLOB wpimath_jni_src
     src/main/native/cpp/jni/WPIMathJNI_DARE.cpp
@@ -196,7 +184,6 @@ target_include_directories(
     wpimath
     PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/main/native/include>
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/protobuf>
         $<INSTALL_INTERFACE:${include_dest}/wpimath>
 )
 
diff --git a/wpimath/src/main/native/cpp/geometry/Pose3d.cpp b/wpimath/src/main/native/cpp/geometry/Pose3d.cpp
index bd7ab8681..be2906259 100644
--- a/wpimath/src/main/native/cpp/geometry/Pose3d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Pose3d.cpp
@@ -9,8 +9,6 @@
 #include <Eigen/Core>
 #include <wpi/json.h>
 
-#include "geometry3d.pb.h"
-
 using namespace frc;
 
 namespace {
diff --git a/wpimath/src/main/native/cpp/geometry/Rotation2d.cpp b/wpimath/src/main/native/cpp/geometry/Rotation2d.cpp
index 69193028f..921e1f81a 100644
--- a/wpimath/src/main/native/cpp/geometry/Rotation2d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Rotation2d.cpp
@@ -8,7 +8,6 @@
 
 #include <wpi/json.h>
 
-#include "geometry2d.pb.h"
 #include "units/math.h"
 
 using namespace frc;
diff --git a/wpimath/src/main/native/cpp/geometry/Rotation3d.cpp b/wpimath/src/main/native/cpp/geometry/Rotation3d.cpp
index 072d023cb..b4dce35a5 100644
--- a/wpimath/src/main/native/cpp/geometry/Rotation3d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Rotation3d.cpp
@@ -13,7 +13,6 @@
 #include <wpi/json.h>
 
 #include "frc/fmt/Eigen.h"
-#include "geometry3d.pb.h"
 #include "units/math.h"
 #include "wpimath/MathShared.h"
 
diff --git a/wpimath/src/main/native/cpp/geometry/Transform2d.cpp b/wpimath/src/main/native/cpp/geometry/Transform2d.cpp
index 157359bfb..25b05907e 100644
--- a/wpimath/src/main/native/cpp/geometry/Transform2d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Transform2d.cpp
@@ -5,7 +5,6 @@
 #include "frc/geometry/Transform2d.h"
 
 #include "frc/geometry/Pose2d.h"
-#include "geometry2d.pb.h"
 
 using namespace frc;
 
diff --git a/wpimath/src/main/native/cpp/geometry/Translation2d.cpp b/wpimath/src/main/native/cpp/geometry/Translation2d.cpp
index 90162018e..842f6c03c 100644
--- a/wpimath/src/main/native/cpp/geometry/Translation2d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Translation2d.cpp
@@ -6,7 +6,6 @@
 
 #include <wpi/json.h>
 
-#include "geometry2d.pb.h"
 #include "units/math.h"
 
 using namespace frc;
diff --git a/wpimath/src/main/native/cpp/geometry/Translation3d.cpp b/wpimath/src/main/native/cpp/geometry/Translation3d.cpp
index c3b8cb4b7..69757f308 100644
--- a/wpimath/src/main/native/cpp/geometry/Translation3d.cpp
+++ b/wpimath/src/main/native/cpp/geometry/Translation3d.cpp
@@ -6,7 +6,6 @@
 
 #include <wpi/json.h>
 
-#include "geometry3d.pb.h"
 #include "units/length.h"
 #include "units/math.h"
 
diff --git a/wpimath/src/main/native/include/frc/controller/ArmFeedforward.h b/wpimath/src/main/native/include/frc/controller/ArmFeedforward.h
index 69091ccd9..4699a50fb 100644
--- a/wpimath/src/main/native/include/frc/controller/ArmFeedforward.h
+++ b/wpimath/src/main/native/include/frc/controller/ArmFeedforward.h
@@ -189,5 +189,4 @@ class WPILIB_DLLEXPORT ArmFeedforward {
 };
 }  // namespace frc
 
-#include "frc/controller/proto/ArmFeedforwardProto.h"
 #include "frc/controller/struct/ArmFeedforwardStruct.h"
diff --git a/wpimath/src/main/native/include/frc/controller/DifferentialDriveWheelVoltages.h b/wpimath/src/main/native/include/frc/controller/DifferentialDriveWheelVoltages.h
index 424cd2d1b..0c9efe1ce 100644
--- a/wpimath/src/main/native/include/frc/controller/DifferentialDriveWheelVoltages.h
+++ b/wpimath/src/main/native/include/frc/controller/DifferentialDriveWheelVoltages.h
@@ -21,5 +21,4 @@ struct DifferentialDriveWheelVoltages {
 
 }  // namespace frc
 
-#include "frc/controller/proto/DifferentialDriveWheelVoltagesProto.h"
 #include "frc/controller/struct/DifferentialDriveWheelVoltagesStruct.h"
diff --git a/wpimath/src/main/native/include/frc/controller/ElevatorFeedforward.h b/wpimath/src/main/native/include/frc/controller/ElevatorFeedforward.h
index 07a0499a5..96bbebcc6 100644
--- a/wpimath/src/main/native/include/frc/controller/ElevatorFeedforward.h
+++ b/wpimath/src/main/native/include/frc/controller/ElevatorFeedforward.h
@@ -196,5 +196,4 @@ class ElevatorFeedforward {
 };
 }  // namespace frc
 
-#include "frc/controller/proto/ElevatorFeedforwardProto.h"
 #include "frc/controller/struct/ElevatorFeedforwardStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Pose2d.h b/wpimath/src/main/native/include/frc/geometry/Pose2d.h
index 65a146c3d..e1831bb9b 100644
--- a/wpimath/src/main/native/include/frc/geometry/Pose2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Pose2d.h
@@ -213,6 +213,5 @@ void from_json(const wpi::json& json, Pose2d& pose);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Pose2dProto.h"
 #include "frc/geometry/struct/Pose2dStruct.h"
 #include "frc/geometry/Pose2d.inc"
diff --git a/wpimath/src/main/native/include/frc/geometry/Pose3d.h b/wpimath/src/main/native/include/frc/geometry/Pose3d.h
index 526dced39..11d084810 100644
--- a/wpimath/src/main/native/include/frc/geometry/Pose3d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Pose3d.h
@@ -215,5 +215,4 @@ void from_json(const wpi::json& json, Pose3d& pose);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Pose3dProto.h"
 #include "frc/geometry/struct/Pose3dStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Quaternion.h b/wpimath/src/main/native/include/frc/geometry/Quaternion.h
index 4fe694cf2..e8a78c744 100644
--- a/wpimath/src/main/native/include/frc/geometry/Quaternion.h
+++ b/wpimath/src/main/native/include/frc/geometry/Quaternion.h
@@ -191,5 +191,4 @@ void from_json(const wpi::json& json, Quaternion& quaternion);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/QuaternionProto.h"
 #include "frc/geometry/struct/QuaternionStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Rotation2d.h b/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
index 6e9cd8e2a..84f3a554c 100644
--- a/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation2d.h
@@ -169,6 +169,5 @@ void from_json(const wpi::json& json, Rotation2d& rotation);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Rotation2dProto.h"
 #include "frc/geometry/struct/Rotation2dStruct.h"
 #include "frc/geometry/Rotation2d.inc"
diff --git a/wpimath/src/main/native/include/frc/geometry/Rotation3d.h b/wpimath/src/main/native/include/frc/geometry/Rotation3d.h
index 98abe007a..d495bb51e 100644
--- a/wpimath/src/main/native/include/frc/geometry/Rotation3d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Rotation3d.h
@@ -195,5 +195,4 @@ void from_json(const wpi::json& json, Rotation3d& rotation);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Rotation3dProto.h"
 #include "frc/geometry/struct/Rotation3dStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Transform2d.h b/wpimath/src/main/native/include/frc/geometry/Transform2d.h
index dccc0e1ee..93ead8cc9 100644
--- a/wpimath/src/main/native/include/frc/geometry/Transform2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Transform2d.h
@@ -124,6 +124,5 @@ class WPILIB_DLLEXPORT Transform2d {
 };
 }  // namespace frc
 
-#include "frc/geometry/proto/Transform2dProto.h"
 #include "frc/geometry/struct/Transform2dStruct.h"
 #include "frc/geometry/Transform2d.inc"
diff --git a/wpimath/src/main/native/include/frc/geometry/Transform3d.h b/wpimath/src/main/native/include/frc/geometry/Transform3d.h
index 94ecfe37e..548fac5fe 100644
--- a/wpimath/src/main/native/include/frc/geometry/Transform3d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Transform3d.h
@@ -130,5 +130,4 @@ class WPILIB_DLLEXPORT Transform3d {
 };
 }  // namespace frc
 
-#include "frc/geometry/proto/Transform3dProto.h"
 #include "frc/geometry/struct/Transform3dStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Translation2d.h b/wpimath/src/main/native/include/frc/geometry/Translation2d.h
index 9d9a77c3b..5063bbfeb 100644
--- a/wpimath/src/main/native/include/frc/geometry/Translation2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Translation2d.h
@@ -214,6 +214,5 @@ void from_json(const wpi::json& json, Translation2d& state);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Translation2dProto.h"
 #include "frc/geometry/struct/Translation2dStruct.h"
 #include "frc/geometry/Translation2d.inc"
diff --git a/wpimath/src/main/native/include/frc/geometry/Translation3d.h b/wpimath/src/main/native/include/frc/geometry/Translation3d.h
index b83b661fd..91887a037 100644
--- a/wpimath/src/main/native/include/frc/geometry/Translation3d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Translation3d.h
@@ -199,6 +199,5 @@ void from_json(const wpi::json& json, Translation3d& state);
 
 }  // namespace frc
 
-#include "frc/geometry/proto/Translation3dProto.h"
 #include "frc/geometry/struct/Translation3dStruct.h"
 #include "frc/geometry/Translation3d.inc"
diff --git a/wpimath/src/main/native/include/frc/geometry/Twist2d.h b/wpimath/src/main/native/include/frc/geometry/Twist2d.h
index d13cc4ec8..498db5669 100644
--- a/wpimath/src/main/native/include/frc/geometry/Twist2d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Twist2d.h
@@ -58,5 +58,4 @@ struct WPILIB_DLLEXPORT Twist2d {
 };
 }  // namespace frc
 
-#include "frc/geometry/proto/Twist2dProto.h"
 #include "frc/geometry/struct/Twist2dStruct.h"
diff --git a/wpimath/src/main/native/include/frc/geometry/Twist3d.h b/wpimath/src/main/native/include/frc/geometry/Twist3d.h
index 3262367dc..e814493d3 100644
--- a/wpimath/src/main/native/include/frc/geometry/Twist3d.h
+++ b/wpimath/src/main/native/include/frc/geometry/Twist3d.h
@@ -78,5 +78,4 @@ struct WPILIB_DLLEXPORT Twist3d {
 };
 }  // namespace frc
 
-#include "frc/geometry/proto/Twist3dProto.h"
 #include "frc/geometry/struct/Twist3dStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/ChassisSpeeds.h b/wpimath/src/main/native/include/frc/kinematics/ChassisSpeeds.h
index f8e844658..859b37204 100644
--- a/wpimath/src/main/native/include/frc/kinematics/ChassisSpeeds.h
+++ b/wpimath/src/main/native/include/frc/kinematics/ChassisSpeeds.h
@@ -259,5 +259,4 @@ struct WPILIB_DLLEXPORT ChassisSpeeds {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/ChassisSpeedsProto.h"
 #include "frc/kinematics/struct/ChassisSpeedsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveKinematics.h b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveKinematics.h
index 0f0cff9fa..bbc4a115c 100644
--- a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveKinematics.h
+++ b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveKinematics.h
@@ -93,5 +93,4 @@ class WPILIB_DLLEXPORT DifferentialDriveKinematics
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/DifferentialDriveKinematicsProto.h"
 #include "frc/kinematics/struct/DifferentialDriveKinematicsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelPositions.h b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelPositions.h
index f22d6874f..1fcb1ec33 100644
--- a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelPositions.h
+++ b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelPositions.h
@@ -50,5 +50,4 @@ struct WPILIB_DLLEXPORT DifferentialDriveWheelPositions {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/DifferentialDriveWheelPositionsProto.h"
 #include "frc/kinematics/struct/DifferentialDriveWheelPositionsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelSpeeds.h b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelSpeeds.h
index 676b8ad59..175a3818c 100644
--- a/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelSpeeds.h
+++ b/wpimath/src/main/native/include/frc/kinematics/DifferentialDriveWheelSpeeds.h
@@ -113,5 +113,4 @@ struct WPILIB_DLLEXPORT DifferentialDriveWheelSpeeds {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/DifferentialDriveWheelSpeedsProto.h"
 #include "frc/kinematics/struct/DifferentialDriveWheelSpeedsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveKinematics.h b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveKinematics.h
index f5a53ca95..c34f135e3 100644
--- a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveKinematics.h
+++ b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveKinematics.h
@@ -194,5 +194,4 @@ class WPILIB_DLLEXPORT MecanumDriveKinematics
 
 }  // namespace frc
 
-#include "frc/kinematics/proto/MecanumDriveKinematicsProto.h"
 #include "frc/kinematics/struct/MecanumDriveKinematicsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelPositions.h b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelPositions.h
index 77d79192b..27015c729 100644
--- a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelPositions.h
+++ b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelPositions.h
@@ -61,5 +61,4 @@ struct WPILIB_DLLEXPORT MecanumDriveWheelPositions {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/MecanumDriveWheelPositionsProto.h"
 #include "frc/kinematics/struct/MecanumDriveWheelPositionsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelSpeeds.h b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelSpeeds.h
index 5df198c80..fe4343466 100644
--- a/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelSpeeds.h
+++ b/wpimath/src/main/native/include/frc/kinematics/MecanumDriveWheelSpeeds.h
@@ -121,5 +121,4 @@ struct WPILIB_DLLEXPORT MecanumDriveWheelSpeeds {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/MecanumDriveWheelSpeedsProto.h"
 #include "frc/kinematics/struct/MecanumDriveWheelSpeedsStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/SwerveModulePosition.h b/wpimath/src/main/native/include/frc/kinematics/SwerveModulePosition.h
index 576bfcf00..c87aee66b 100644
--- a/wpimath/src/main/native/include/frc/kinematics/SwerveModulePosition.h
+++ b/wpimath/src/main/native/include/frc/kinematics/SwerveModulePosition.h
@@ -43,5 +43,4 @@ struct WPILIB_DLLEXPORT SwerveModulePosition {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/SwerveModulePositionProto.h"
 #include "frc/kinematics/struct/SwerveModulePositionStruct.h"
diff --git a/wpimath/src/main/native/include/frc/kinematics/SwerveModuleState.h b/wpimath/src/main/native/include/frc/kinematics/SwerveModuleState.h
index 3060d35fc..b24b86b83 100644
--- a/wpimath/src/main/native/include/frc/kinematics/SwerveModuleState.h
+++ b/wpimath/src/main/native/include/frc/kinematics/SwerveModuleState.h
@@ -48,5 +48,4 @@ struct WPILIB_DLLEXPORT SwerveModuleState {
 };
 }  // namespace frc
 
-#include "frc/kinematics/proto/SwerveModuleStateProto.h"
 #include "frc/kinematics/struct/SwerveModuleStateStruct.h"
diff --git a/wpimath/src/main/native/include/frc/system/plant/DCMotor.h b/wpimath/src/main/native/include/frc/system/plant/DCMotor.h
index a26794524..164de6047 100644
--- a/wpimath/src/main/native/include/frc/system/plant/DCMotor.h
+++ b/wpimath/src/main/native/include/frc/system/plant/DCMotor.h
@@ -249,5 +249,4 @@ class WPILIB_DLLEXPORT DCMotor {
 
 }  // namespace frc
 
-#include "frc/system/plant/proto/DCMotorProto.h"
 #include "frc/system/plant/struct/DCMotorStruct.h"
diff --git a/wpimath/src/main/native/include/frc/trajectory/Trajectory.h b/wpimath/src/main/native/include/frc/trajectory/Trajectory.h
index 753bf9e2d..e103d83fa 100644
--- a/wpimath/src/main/native/include/frc/trajectory/Trajectory.h
+++ b/wpimath/src/main/native/include/frc/trajectory/Trajectory.h
@@ -145,6 +145,3 @@ WPILIB_DLLEXPORT
 void from_json(const wpi::json& json, Trajectory::State& state);
 
 }  // namespace frc
-
-#include "frc/trajectory/proto/TrajectoryProto.h"
-#include "frc/trajectory/proto/TrajectoryStateProto.h"
diff --git a/wpiutil/CMakeLists.txt b/wpiutil/CMakeLists.txt
index b4482ff04..aecf8f2af 100644
--- a/wpiutil/CMakeLists.txt
+++ b/wpiutil/CMakeLists.txt
@@ -157,7 +157,7 @@ if(MSVC)
     target_compile_definitions(wpiutil PRIVATE -D_CRT_SECURE_NO_WARNINGS)
 endif()
 wpilib_target_warnings(wpiutil)
-target_link_libraries(wpiutil protobuf::libprotobuf Threads::Threads ${CMAKE_DL_LIBS})
+target_link_libraries(wpiutil Threads::Threads ${CMAKE_DL_LIBS})
 
 if(ATOMIC)
     target_link_libraries(wpiutil ${ATOMIC})
diff --git a/wpiutil/src/main/native/include/wpi/DataLog.h b/wpiutil/src/main/native/include/wpi/DataLog.h
index 6187662c0..2057ba81e 100644
--- a/wpiutil/src/main/native/include/wpi/DataLog.h
+++ b/wpiutil/src/main/native/include/wpi/DataLog.h
@@ -26,7 +26,6 @@
 #include "wpi/StringMap.h"
 #include "wpi/condition_variable.h"
 #include "wpi/mutex.h"
-#include "wpi/protobuf/Protobuf.h"
 #include "wpi/struct/Struct.h"
 #include "wpi/timestamp.h"
 #endif  // __cplusplus
@@ -239,26 +238,6 @@ class DataLog final {
         timestamp);
   }
 
-  /**
-   * Registers a protobuf schema. Duplicate calls to this function with the same
-   * name are silently ignored.
-   *
-   * @tparam T protobuf serializable type
-   * @param msg protobuf message
-   * @param timestamp Time stamp (0 to indicate now)
-   */
-  template <ProtobufSerializable T>
-  void AddProtobufSchema(ProtobufMessage<T>& msg, int64_t timestamp = 0) {
-    if (timestamp == 0) {
-      timestamp = Now();
-    }
-    msg.ForEachProtobufDescriptor(
-        [this](auto typeString) { return HasSchema(typeString); },
-        [this, timestamp](auto typeString, auto schema) {
-          AddSchema(typeString, "proto:FileDescriptorProto", schema, timestamp);
-        });
-  }
-
   /**
    * Registers a struct schema. Duplicate calls to this function with the same
    * name are silently ignored.
@@ -1062,44 +1041,6 @@ class StructArrayLogEntry : public DataLogEntry {
   [[no_unique_address]] std::tuple<I...> m_info;
 };
 
-/**
- * Log protobuf serializable objects.
- */
-template <ProtobufSerializable T>
-class ProtobufLogEntry : public DataLogEntry {
-  using P = Protobuf<T>;
-
- public:
-  ProtobufLogEntry() = default;
-  ProtobufLogEntry(DataLog& log, std::string_view name, int64_t timestamp = 0)
-      : ProtobufLogEntry{log, name, {}, timestamp} {}
-  ProtobufLogEntry(DataLog& log, std::string_view name,
-                   std::string_view metadata, int64_t timestamp = 0) {
-    m_log = &log;
-    log.AddProtobufSchema<T>(m_msg, timestamp);
-    m_entry = log.Start(name, m_msg.GetTypeString(), metadata, timestamp);
-  }
-
-  /**
-   * Appends a record to the log.
-   *
-   * @param data Data to record
-   * @param timestamp Time stamp (may be 0 to indicate now)
-   */
-  void Append(const T& data, int64_t timestamp = 0) {
-    SmallVector<uint8_t, 128> buf;
-    {
-      std::scoped_lock lock{m_mutex};
-      m_msg.Pack(buf, data);
-    }
-    m_log->AppendRaw(m_entry, buf, timestamp);
-  }
-
- private:
-  wpi::mutex m_mutex;
-  ProtobufMessage<T> m_msg;
-};
-
 }  // namespace wpi::log
 
 extern "C" {
diff --git a/wpiutil/wpiutil-config.cmake.in b/wpiutil/wpiutil-config.cmake.in
index 3926ecb91..7cad0f945 100644
--- a/wpiutil/wpiutil-config.cmake.in
+++ b/wpiutil/wpiutil-config.cmake.in
@@ -2,7 +2,6 @@ include(CMakeFindDependencyMacro)
 @FILENAME_DEP_REPLACE@
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_dependency(Threads)
-find_dependency(Protobuf)
 @FMTLIB_SYSTEM_REPLACE@
 
 if(@USE_SYSTEM_FMTLIB@)
